src/js/helpers/timeline/declarationHelper.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2020 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DataSet } from 'vis-timeline';
+
+const hasSubtasks = (node) => !!node.subtasks;
+
+const hasProcessedTransition = (node) => !!node.on && !!node.response.transition && !!node.on[node.response.transition];
+
+const getTransition = (node) => node.on[node.response.transition];
+
+export const getProcessedNodes = (root) => (hasProcessedTransition(root)
+  ? [root, ...getProcessedNodes(getTransition(root))]
+  : [root]);
+
+const generateGroupName = (node) => node.label;
+
+const generateSubgroupData = (parent) => (parent ? { subgroupOf: generateGroupName(parent) } : {});
+
+const createTimelineItem = (node, parent) => ({
+  id: node.id,
+  start: parseInt(node.started, 10),
+  end: parseInt(node.finished, 10),
+  content: '',
+  group: generateGroupName(node),
+  ...generateSubgroupData(parent),
+});
+
+export const createTimelineItems = (node, parent = null) => (hasSubtasks(node)
+  ? [
+    createTimelineItem(node, parent),
+    ...node.subtasks
+      .flatMap((subtask) => getProcessedNodes(subtask))
+      .flatMap((subtask) => createTimelineItems(subtask, node))]
+  : [
+    createTimelineItem(node, parent),
+  ]);
+
+export const reduceSubgroups = (prev, { group, subgroupOf }) => {
+  const total = { ...prev };
+
+  if (subgroupOf) {
+    total[subgroupOf] = [...total[subgroupOf], group];
+  }
+
+  total[group] = total[group] || [];
+
+  return total;
+};
+
+export const extractGroupData = (items) => Object.entries(items
+  .map(({ group, subgroupOf }) => ({ group, subgroupOf }))
```

Comment:
What does it do?

Reply:
It does practically nothing, must have slipped under my radar ;)